src/main/java/frc/robot/subsystems/ControlPanel.java

```diff
@@ -1,36 +1,196 @@
+        detectColor();
+    }
+
+    // Mapping the cases to the correct color due to the difference in what vision
+    // is reading and what the field is reading
+    public ColorStates getTargetColor() {
+        gameData = DriverStation.getInstance().getGameSpecificMessage();
+        if (gameData.length() > 0) {
+            switch (gameData.charAt(0)) {
+            case 'R':
+                color = ColorStates.BLUE;
+                break;
+            case 'Y':
+                color = ColorStates.GREEN;
+                break;
+            case 'B':
+                color = ColorStates.RED;
+                break;
+            case 'G':
+                color = ColorStates.YELLOW;
+                break;
+            default:
+                color = ColorStates.OTHER;
+                break;
+            }
+        } else {
+            color = ColorStates.OTHER;
+        }
+        return color;
+    }
+
+    public CommandBase spinControlPanel(DoubleSupplier speed) {
+        CommandBase cmd = new FunctionalCommand(() -> {
+
         }, () -> {
+            spinnerMotor.set(speed.getAsDouble());
+        }, (interrupted) -> {
             spinnerMotor.stopMotor();
-        }, this);
+        }, () -> false, this);
+        cmd.setName("Spin Control Panel");
+        return cmd;
+    }
+
+    public ColorStates detectColor() {
+
+        Color detectedColor = m_colorSensor.getColor();
+
+        ColorStates colorString;
+        ColorMatchResult match = m_colorMatcher.matchClosestColor(detectedColor);
+
+        if (match.color == kBlueTarget) {
+            colorString = ColorStates.BLUE;
+        } else if (match.color == kRedTarget) {
+            colorString = ColorStates.RED;
+        } else if (match.color == kGreenTarget) {
+            colorString = ColorStates.GREEN;
+        } else if (match.color == kYellowTarget) {
+            colorString = ColorStates.YELLOW;
+        } else {
+            colorString = ColorStates.OTHER;
+        }
+
+        /**
+         * Open Smart Dashboard or Shuffleboard to see the color detected by the sensor.
+         */
+        SmartDashboard.putNumber("Red", detectedColor.red);
+        SmartDashboard.putNumber("Green", detectedColor.green);
+        SmartDashboard.putNumber("Blue", detectedColor.blue);
+        SmartDashboard.putNumber("Confidence", match.confidence);
+        SmartDashboard.putString("Detected Color", colorString.name());
+        return colorString;
+
     }
-}
\ No newline at end of file
+
+    public CommandBase stageTwoRotation() {
+        return new CommandBase() {
+            int i = 0;
+            ColorStates firstColor = detectColor();
+
+            @Override
+            public void initialize() {
+                firstColor = detectColor();
+                super.initialize();
+                spinnerMotor.set(spinnerMotorSpeed);
+            }
+
+            @Override
+            public boolean isFinished() {
+                return 40 < i && i < 50;
```

Comment:
`i` will never get to 50 here, no need to have the extra check.

Reply:
fixed